app/code/Magento/Catalog/Model/Indexer/Category/Flat/AbstractAction.php

```diff
@@ -77,20 +76,31 @@ class AbstractAction
      */
     protected $skipStaticColumns = [];
 
+    /**
+     * @var SkipStaticColumnsProvider
+     */
+    private $staticColumnsProvider;
+
     /**
      * @param ResourceConnection $resource
      * @param StoreManagerInterface $storeManager
      * @param Helper $resourceHelper
+     * @param MetadataPool|null $categoryMetadata
+     * @param SkipStaticColumnsProvider|null $provider
      */
     public function __construct(
         ResourceConnection $resource,
         StoreManagerInterface $storeManager,
-        Helper $resourceHelper
+        Helper $resourceHelper,
+        MetadataPool $categoryMetadata = null,
+        SkipStaticColumnsProvider $provider = null
     ) {
         $this->resource = $resource;
         $this->connection = $resource->getConnection();
         $this->storeManager = $storeManager;
         $this->resourceHelper = $resourceHelper;
+        $this->categoryMetadata = $categoryMetadata ?: ObjectManager::getInstance()->get(MetadataPool::class);
+        $this->staticColumnsProvider = $provider ?: ObjectManager::getInstance()->get(SkipStaticColumnsProvider::class);
```

Comment:
@ihor-sviziev I checked dependencies and preference for this classes. I think they are not so heavy. I removed methods that use an  ObjectManager for Static tests. For this two classes we can add proxy definition to di.xml What do you think? Thanks!

Reply:
If the dependencies are not heavy - we can keep it as is w/o adding a proxy class.